### PR TITLE
HDDS-3861. Fix handlePipelineFailure throw exception if role is follower

### PR DESCRIPTION
**Pull request by runzhiwang**
## What changes were proposed in this pull request?
1. if role is follower, ozone will throw exception.
image

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3861

## How was this patch tested?

Existed tests

### CODE REVIEW

**Comment by runzhiwang** [comment]
@bshashikant Could you help review this patch ? Thank you very much.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java [hunk 1]
```diff
@@ -552,6 +552,10 @@ private void handlePipelineFailure(RaftGroupId groupId,
       msg = datanode + " is in candidate state for " +
           roleInfoProto.getCandidateInfo().getLastLeaderElapsedTimeMs() + "ms";
       break;
+    case FOLLOWER:
+      msg = datanode + " is in follower state for " +
```
Comment (bshashikant) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java:556]: Can we update the msg accordingly with the reason. The reason precisely was when installSnapshotFromLeader gets invoked on follower, as leader snapshot doesn't contain any data to replay. All the log entries prior to the snapshot might have been purged. So , follower should not try to install snapshot from leader but can close the pipeline here.

Comment (runzhiwang) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java]: @bshashikant Thanks for you review and explanation. I have updated the patch.

**Comment by runzhiwang** [comment]
@adoroszlai  Could you help merge it ?  The comment is just about the message of reason, I have updated it, I think it does not matter. But if without this PR, the cluster is very unstable.

**Pull request merged**